src/Database/Schema/TableSchemaInterface.php

```diff
@@ -241,9 +241,9 @@ public function getConstraint(string $name): ?array;
      * Remove a constraint.
      *
      * @param string $name Name of the constraint to remove
-     * @return \Cake\Database\Schema\TableSchemaInterface
+     * @return self
```

Comment:
`static` then AFAIK, since those are chainable use and as such extended classes can have added methods on top.